### ISSUE TITLE
检测模型和识别模型根本跑不起来的问题，参考很多相关文章，也实验了很多次就是不行啊，，，

### ISSUE THREAD

**Issue by kognyouze123** [PaddlePaddle/PaddleOCR#9687]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：paddle_gpu=2.4.2 paddleOCR_release:2.1/2.5都试了
- 版本号/Version：Paddle：  PaddleOCR： 问题相关组件/Related components：
- 运行指令/Command Code：python tools/train.py -c configs/det/det_mv3_db.yml -o Global.pretrain_weights=./pretrain_models/ch_ppocr_server_v2.0_det_train/best_accuracy > t
rain_det.log
- 完整报错/Complete Error Message：[2023-04-10 15:55:22,204] [    INFO] utility.py:35 -                 mean : [0.485, 0.456, 0.406]
[2023-04-10 15:55:22,204] [    INFO] utility.py:35 -                 order : hwc
[2023-04-10 15:55:22,204] [    INFO] utility.py:35 -                 scale : 1./255.
[2023-04-10 15:55:22,204] [    INFO] utility.py:35 -                 std : [0.229, 0.224, 0.225]
[2023-04-10 15:55:22,205] [    INFO] utility.py:35 -             ToCHWImage : None
[2023-04-10 15:55:22,205] [    INFO] utility.py:28 -             KeepKeys :
[2023-04-10 15:55:22,205] [    INFO] utility.py:35 -                 keep_keys : ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask']
[2023-04-10 15:55:22,205] [    INFO] utility.py:28 -     loader : 
[2023-04-10 15:55:22,206] [    INFO] utility.py:35 -         batch_size_per_card : 16
  File "tools/train.py", line 100, in main
    program.train(config, train_dataloader, valid_dataloader, device, model,
  File "D:\Anaconda3\envs\ocr0.1\Lib\site-packages\paddle\PaddleOCR-release-2.1\tools\program.py", line 216, in train
    loss = loss_class(preds, batch)
  File "D:\Anaconda3\envs\paddle_train\lib\site-packages\paddle\fluid\dygraph\layers.py", line 1012, in __call__
    return self.forward(*inputs, **kwargs)
  File "D:\Anaconda3\envs\ocr0.1\Lib\site-packages\paddle\PaddleOCR-release-2.1\ppocr\losses\det_db_loss.py", line 61, in forward
    loss_binary_maps = self.dice_loss(binary_maps, label_shrink_map,
  File "D:\Anaconda3\envs\paddle_train\lib\site-packages\paddle\fluid\dygraph\layers.py", line 1012, in __call__
    return self.forward(*inputs, **kwargs)
  File "D:\Anaconda3\envs\ocr0.1\Lib\site-packages\paddle\PaddleOCR-release-2.1\ppocr\losses\det_basic_loss.py", line 131, in forward
    assert loss <= 1
AssertionError

根本没跑起来。。试了改参数，num_work 0，batch_size从0-8都试了，现在是16

- 然后log一直卡在这里：
[2023/04/10 15:55:22] root INFO:                 size : [640, 640]
[2023/04/10 15:55:22] root INFO:             MakeBorderMap : 
[2023/04/10 15:55:22] root INFO:                 shrink_ratio : 0.4
[2023/04/10 15:55:22] root INFO:                 thresh_max : 0.7
[2023/04/10 15:55:22] root INFO:                 thresh_min : 0.3
[2023/04/10 15:55:22] root INFO:             MakeShrinkMap : 
[2023/04/10 15:55:22] root INFO:                 min_text_size : 8
[2023/04/10 15:55:22] root INFO:                 shrink_ratio : 0.4
[2023/04/10 15:55:22] root INFO:             NormalizeImage : 
[2023/04/10 15:55:22] root INFO:                 mean : [0.485, 0.456, 0.406]
[2023/04/10 15:55:22] root INFO:                 order : hwc
[2023/04/10 15:55:22] root INFO:                 scale : 1./255.
[2023/04/10 15:55:22] root INFO:                 std : [0.229, 0.224, 0.225]
[2023/04/10 15:55:22] root INFO:             ToCHWImage : None
[2023/04/10 15:55:22] root INFO:             KeepKeys : 
[2023/04/10 15:55:22] root INFO:                 keep_keys : ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask']
[2023/04/10 15:55:22] root INFO:     loader : 
[2023/04/10 15:55:22] root INFO:         batch_size_per_card : 16
[2023/04/10 15:55:22] root INFO:         drop_last : False
[2023/04/10 15:55:22] root INFO:         num_workers : 0
[2023/04/10 15:55:22] root INFO:         shuffle : True
[2023/04/10 15:55:22] root INFO:         use_shared_memory : True
[2023/04/10 15:55:22] root INFO: train with paddle 2.4.2 and device Place(gpu:0)
[2023/04/10 15:55:22] root INFO: Initialize indexs of datasets:['C:/Users/qy/Desktop/PaddleOCR-release-2.5/train_data/det/train.txt']
[2023/04/10 15:55:22] root INFO: Initialize indexs of datasets:['C:/Users/qy/Desktop/PaddleOCR-release-2.5/train_data/det/val.txt']
[2023/04/10 15:55:23] root INFO: load pretrained model from ['D:/Anaconda3/envs/ocr0.1/Lib/site-packages/paddle/PaddleOCR-release-2.1/pretrain_models/ch_ppocr_server_v2.0_det_train/best_accuracy']
[2023/04/10 15:55:23] root INFO: train dataloader has 14 iters
[2023/04/10 15:55:23] root INFO: valid dataloader has 71 iters
[2023/04/10 15:55:23] root INFO: During the training process, after the 0th iteration, an evaluation is run every 2000 iterations
[2023/04/10 15:55:23] root INFO: Initialize indexs of datasets:['C:/Users/qy/Desktop/PaddleOCR-release-2.5/train_data/det/train.txt']

请教各位大神，这个问题该如何解决呀，拜谢

**Comment by LDOUBLEV** [comment]
2.1的分支太旧了，都不维护了，建议用2.6分支吧；
我这边在icdar2015上跑是正常的，报错提示出现了loss>1的情况，你有修改loss部分的代码吗？

File "D:\Anaconda3\envs\ocr0.1\Lib\site-packages\paddle\PaddleOCR-release-2.1\ppocr\losses\det_basic_loss.py", line 131, in forward
assert loss <= 1
AssertionError

**Comment by kognyouze123** [comment]
> 
请问是要修改源码嘛，，，这个没修改过的，，，
.现在看了一些回复感觉是不是和我数据有关系？我是业务需要把数字写成特殊字符作为结果训练了。不知道这样会不会产生这个问题？

**Comment by LDOUBLEV** [comment]
不用修改代码就能跑啊，可能是你的数据导致的，你检查下为什么计算的iou loss会出现>1的情况吧

**Comment by kognyouze123** [comment]
> 不用修改代码就能跑啊，可能是你的数据导致的，你检查下为什么计算的iou loss会出现>1的情况吧

请问这个怎么检查。。。debug到这一步就不能往下执行了。。

**Comment by kognyouze123** [comment]
> 不用修改代码就能跑啊，可能是你的数据导致的，你检查下为什么计算的iou loss会出现>1的情况吧

请问我现在尝试在train.txt里把原来PPLable打标签的300条数据分割成10条每次，使用-o Global.pretrain_weights训练（另外看到官方文档断点训练是Global.checkpoints，不知道还使用Global.pretrain_weights可以不）就一直显示ppocr INFO: save model in ./output/db_mv3/latest 看起来是训练上了吧
